financial-connections/src/main/java/com/stripe/android/financialconnections/domain/IsLinkWithStripe.kt

```diff
@@ -0,0 +1,16 @@
+package com.stripe.android.financialconnections.domain
+
+import javax.inject.Inject
+
+internal fun interface IsLinkWithStripe {
+    suspend operator fun invoke(): Boolean
+}
+
+internal class RealIsLinkWithStripe @Inject constructor(
+    private val getOrFetchSync: GetOrFetchSync,
+) : IsLinkWithStripe {
+
+    override suspend operator fun invoke(): Boolean {
+        return getOrFetchSync().manifest.isLinkWithStripe ?: false
+    }
+}
```

Comment:
I'm fine adding this interactor, but maybe it obscures a bit that we're retrieving (maybe re-fetching) `synchronize`? we can alternatively just inject `GetOrFetchSync` wherever needed and have a `SynchronizeResponse` / `Manifest` extension function. 

Happy to keep it as is if you don't see it that way!

Reply:
That’s a fair concern. My initial thinking was that injecting something simple (like `IsLinkWithStripe`) means that we can easily mock it in unit tests, by just writing `isLinkWithStripe = { true }`.

We do have decent test fixtures for the `Manifest`, so I can switch this over to `GetOrFetchSync`.

Reply:
Actually using `initialState.isLinkWithStripe` now inside `IsLinkWithStripe`, meaning that it will never perform any network requests.